vmdk_plugin/refcnt.go

```diff
@@ -252,7 +343,11 @@ func (r refCountsMap) discoverAndSync(c *client.Client, d *vmdkDriver) error {
 
 // syncronize mount info with refcounts - and unmounts if needed
 func (r refCountsMap) syncMountsWithRefCounters(d *vmdkDriver) {
-	for vol, cnt := range r {
+	// Lock the refCountsMap
+	r.mtx.Lock()
+	defer r.mtx.Unlock()
+
+	for vol, cnt := range r.refMap {
```

Comment:
Yes, attention will be needed when touching the init functions in the future. If the online recovery ideia is unlikely we should remove the locks then.